File: strings/java/src/Transacoes.java
import java.util.ArrayList;
import java.util.List;

public class Transacoes {

	public static void main(String[] args) {
		System.out.println(invalidTransactions(new String[] { "alice,20,800,mtv", "alice,50,100,beijing" }));
		System.out.println(invalidTransactions(new String[] { "alice,20,800,mtv", "alice,50,1200,mtv" }));
		System.out.println(invalidTransactions(new String[] { "alice,20,800,mtv", "bob,50,1200,mtv" }));
	}

	public static List<String> invalidTransactions(String[] transactions) {

		class Transaction {
			String name;
			int time;
			int money;
			String city;

			public Transaction(String csv) {
				String[] parts = csv.split(",");
				name = parts[0];
				time = Integer.parseInt(parts[1]);
				money = Integer.parseInt(parts[2]);
				city = parts[3];
			}
		}

		List<String> result = new ArrayList<>();
		boolean[] invalidArray = new boolean[transactions.length];

		for (int i = 0; i < transactions.length; i++) {

			Transaction ti = new Transaction(transactions[i]);

			if (ti.money > 1000) {
				invalidArray[i] = true;
			}

			for (int j = i + 1; j < transactions.length; j++) {

				Transaction tj = new Transaction(transactions[j]);

				if (ti.name.equals(tj.name) && Math.abs(ti.time - tj.time) <= 60 && !ti.city.equals(tj.city)) {
					invalidArray[i] = true;
					invalidArray[j] = true;
				}
			}
		}

		for (int i = 0; i < transactions.length; i++) {
			if (invalidArray[i])
				result.add(transactions[i]);
		}
		return result;
	}
}

File: strings/python/transacoes.py
def invalid_transactions(transactions):

    class Transaction:
        def __init__(self, csv):
            parts = csv.split(",")
            self.name = parts[0]
            self.time = int(parts[1])
            self.money = int(parts[2])
            self.city = parts[3]

    result = []
    invalid_array = [False] * len(transactions)

    for i in range(len(transactions)):
        ti = Transaction(transactions[i])

        if ti.money > 1000:
            invalid_array[i] = True

        for j in range(i + 1, len(transactions)):
            tj = Transaction(transactions[j])

            if ti.name == tj.name and abs(ti.time - tj.time) <= 60 and ti.city != tj.city:
                invalid_array[i] = True
                invalid_array[j] = True

    for i in range(len(transactions)):
        if invalid_array[i]:
            result.append(transactions[i])

    return result


print(invalid_transactions(["alice,20,800,mtv", "alice,50,100,beijing"]))
print(invalid_transactions(["alice,20,800,mtv", "alice,50,1200,mtv"]))
print(invalid_transactions(["alice,20,800,mtv", "bob,50,1200,mtv"]))
